File: tests/integration-tests/src/test/java/com/jaynewstrom/json/sample/extraneous/ExtraneousTest.java
package com.jaynewstrom.json.sample.extraneous;

import com.jaynewstrom.json.sample.JsonTestHelper;

import org.junit.Test;

import static org.fest.assertions.api.Assertions.assertThat;

public final class ExtraneousTest {
    @Test public void testDeserializer_whenThereIsAnExtraneousObjectInJson() {
        JsonTestHelper testHelper = new JsonTestHelper();
        String json = "{\"extra\":{\"extra\":{\"readAllAboutIt\":{}}},\"foo\":\"bar\",\"extra2\":{}}";
        Extraneous extraneous = testHelper.deserializeString(Extraneous.class, json);
        assertThat(extraneous.foo).isEqualTo("bar");
    }

    @Test public void testDeserializer_whenThereIsAnExtraneousArrayInJson() {
        JsonTestHelper testHelper = new JsonTestHelper();
        String json = "{\"extra\":[{\"extra\":[\"readAllAboutIt\"]}],\"foo\":\"bar\",\"extra2\":[]}";
        Extraneous extraneous = testHelper.deserializeString(Extraneous.class, json);
        assertThat(extraneous.foo).isEqualTo("bar");
    }

    @Test public void testDeserializer_whenThereIsAnExtraneousField() {
        JsonTestHelper testHelper = new JsonTestHelper();
        String json = "{\"extra\":\"blah\",\"foo\":\"bar\",\"extra2\":false}";
        Extraneous extraneous = testHelper.deserializeString(Extraneous.class, json);
        assertThat(extraneous.foo).isEqualTo("bar");
    }
}

File: tests/integration-tests/src/test/java/com/jaynewstrom/jsonDelight/sample/extraneous/ExtraneousTest.kt
package com.jaynewstrom.jsonDelight.sample.extraneous

import com.jaynewstrom.jsonDelight.sample.JsonTestHelper
import org.fest.assertions.api.Assertions.assertThat
import org.junit.Test

class ExtraneousTest {
    @Test fun testDeserializer_whenThereIsAnExtraneousObjectInJson() {
        val testHelper = JsonTestHelper()
        val json = "{\"extra\":{\"extra\":{\"readAllAboutIt\":{}}},\"foo\":\"bar\",\"extra2\":{}}"
        val (foo) = testHelper.deserializeString(Extraneous::class.java, json)
        assertThat(foo).isEqualTo("bar")
    }

    @Test fun testDeserializer_whenThereIsAnExtraneousArrayInJson() {
        val testHelper = JsonTestHelper()
        val json = "{\"extra\":[{\"extra\":[\"readAllAboutIt\"]}],\"foo\":\"bar\",\"extra2\":[]}"
        val (foo) = testHelper.deserializeString(Extraneous::class.java, json)
        assertThat(foo).isEqualTo("bar")
    }

    @Test fun testDeserializer_whenThereIsAnExtraneousField() {
        val testHelper = JsonTestHelper()
        val json = "{\"extra\":\"blah\",\"foo\":\"bar\",\"extra2\":false}"
        val (foo) = testHelper.deserializeString(Extraneous::class.java, json)
        assertThat(foo).isEqualTo("bar")
    }
}
